tests/base.py

```diff
@@ -53,23 +44,54 @@ def start_run(self, run_id):
         )
         self.assertEqual(response.code, OK_REQUEST_CODE)
 
-    def poll_job(self, worker_id):
-        headers = {api_key.AUTH: self.token, api_key.WORKER_ID: worker_id}
+    def register_worker(self):
+        headers = {api_key.AUTH: self.token}
+
+        response = self.fetch(self.get_url("{}/{}".format(WORKER_REGISTER_ENDPOINT, "mock_hostname")), method='GET',
+                              headers=headers, body=None)
+        self.assertEqual(response.code, OK_REQUEST_CODE)
+        response_body = json.loads(response.body)
+        self.assertIn(api_key.WORKER_ID, response_body["data"])
+        return response_body["data"].get(api_key.WORKER_ID)
+
+    def poll_job(self, worker_id, empty_job=False):
+        headers = {api_key.AUTH: self.token}
 
         response = self.fetch(
-            self.get_url(GRADING_JOB_ENDPOINT), method='GET', headers=headers, body=None
+            self.get_url("{}/{}".format(GRADING_JOB_ENDPOINT, worker_id)), method='GET', headers=headers, body=None
         )
-        self.assertEqual(response.code, OK_REQUEST_CODE)
+
+        self.assertEqual(response.code, QUEUE_EMPTY_CODE if empty_job else OK_REQUEST_CODE)
+        response_body = json.loads(response.body)
+        self.assertIn(api_key.JOB_ID, response_body["data"])
+        self.assertIn(api_key.STAGES, response_body["data"])
+        return response_body["data"]
+
+    def safe_poll_job(self, worker_id):
+        headers = {api_key.AUTH: self.token}
+
+        while True:
+            response = self.fetch(
+                self.get_url("{}/{}".format(GRADING_JOB_ENDPOINT, worker_id)), method='GET', headers=headers, body=None
+            )
```

Comment:
This is going to eventually destroy the CPU with HTTP requests. Could we get a sleep or some condition wakeup?

Reply:
Agree. In its current usage it always succeds in less than 2 iterations. But this function would not be reusable if I assume the above. Will add a sleep.

Reply:
Should be fixed